Gems/Atom/Feature/Common/Code/Include/Atom/Feature/RayTracing/RayTracingPass.h

```diff
@@ -7,21 +7,158 @@
  */
 #pragma once
 
+#include <Atom/Feature/RayTracing/RayTracingFeatureProcessorInterface.h>
 #include <Atom/RHI/DispatchRaysIndirectBuffer.h>
 #include <Atom/RHI/DispatchRaysItem.h>
 #include <Atom/RHI/RayTracingPipelineState.h>
 #include <Atom/RHI/RayTracingShaderTable.h>
 #include <Atom/RPI.Public/Pass/RenderPass.h>
 #include <Atom/RPI.Public/Shader/Shader.h>
 #include <Atom/RPI.Public/Shader/ShaderReloadNotificationBus.h>
+#include <AtomCore/std/containers/small_vector.h>
 #include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/std/containers/fixed_vector.h>
 
 namespace AZ
 {
     namespace Render
     {
         struct RayTracingPassData;
 
+        // One ShaderLib corresponds to one compiled .shader - file with Raytracing shader functions, but one .shader can contain up to five
+        // different functions. A HitGroup specifies which ClosestHit, AnyHit and Intersection - shader functions are in the same Hit-Group,
+        // but they don't have to be from the same ShaderLib.
+        //
+        // the RayTracingShaderLibs class holds the ShaderLibs, and the RayTracingHitGroups class holds the Hit-Groups, but the logic to
+        // create the Hit-Groups resides in the function RayTracingPass::PrepareHitGroups(), and is somewhat hard-coded for our specific use
+        // case.
+
+        // Loaded Raytracing Shader-Libraries
+        class RayTracingShaderLibs
```

Comment:
Why is this added to RayTracingPass? Have you looked at DiffuseProbegrid gem to see how it handles loading shaders and ShaderLibrary?